Source/WebCore/Modules/webxr/WebXRSession.cpp

```diff
@@ -754,7 +754,10 @@ bool WebXRSession::isHandTrackingEnabled() const
 // https://immersive-web.github.io/hit-test/#dom-xrsession-requesthittestsource
 void WebXRSession::requestHitTestSource(const XRHitTestOptionsInit& init, RequestHitTestSourcePromise&& promise)
 {
-    // 3. If session’s ended value is true, throw an InvalidStateError and abort these steps.
+    if (!m_requestedFeatures.contains(PlatformXR::SessionFeature::HitTest)) {
+        promise.reject(Exception { ExceptionCode::NotSupportedError });
+        return;
+    }
```

Comment:
Instead of removing the comment, I think it would be better to move it below the feature check:

```suggestion
    }
    // 3. If session’s ended value is true, throw an InvalidStateError and abort these steps.
```

Reply:
We're in the process of removing comments even if they're verbatim copies of the spec.  As you know they get rotten easily and in WebKit are discouraged unless the code is not self-explanatory or requires digging into other files to be understandable.